Mathlib/Tactic/Linter/FlexibleLinter.lean

```diff
@@ -442,22 +477,43 @@ def flexibleLinter : Linter where run := withSetOptionIn fun _stx => do
               for d in st.toFMVarId currMv0 lctx0 do
                 if !foundFvs.contains d then foundFvs := foundFvs.insert d
             for l in foundFvs do
-              if let some (_stdLoc, (st, kind)) := stains.find? (Prod.fst · == l) then
-                msgs := msgs.push (s, kind, st)
+              if let some (_stdLoc, stainData) := stains.find? (Prod.fst · == l) then
+                msgs := msgs.push (s, stainData)
 
       -- tactics often change the name of the current `MVarId`, so we migrate the `FvarId`s
       -- in the "old" `mvars` to the "same" `FVarId` in the "new" `mvars`
-      let mut new : Array ((FVarId × MVarId) × (Stained × Syntax)) := .empty
-      for (fv, (stLoc, kd)) in stains do
+      let mut new : Array ((FVarId × MVarId) × StainData) := .empty
+      for (fv, stainData) in stains do
         let psisted := reallyPersist #[fv] mvs0 mvs1 ctx0 ctx1
         if psisted == #[] && mvs1 != [] then
-          new := new.push (fv, (stLoc, kd))
-          dbg_trace "lost {((fv.1.name, fv.2.name), stLoc, kd)}"
-        for p in psisted do new := new.push (p, (stLoc, kd))
+          new := new.push (fv, stainData)
+          dbg_trace "lost {((fv.1.name, fv.2.name), stainData.stained, stainData.stx)}"
+        for p in psisted do new := new.push (p, stainData)
       stains := new
 
-  for (s, stainStx, d) in msgs do
+  for (s, stainData) in msgs do
+    let stainStx := stainData.stx
+    let d := stainData.stained
     let stainStr := (stainStx.reprint.getD s!"{stainStx}").trim
+    -- Try to generate a "simp only [...]" suggestion by re-running simp
+    let suggestion? ← match stainStx.getKind with
+      | ``Lean.Parser.Tactic.simp | ``Lean.Parser.Tactic.simpAll => try
```

Comment:
We have two `match stainStx.getKind` in a row in an already complicated definition. I would prefer to have the `Try this: simp only [...]` generation as a separate function that we call from the `match` below.

Reply:
I've addressed this now.